### ISSUE TITLE
Gaussian Sampler doesn't Build for QEMU

### ISSUE THREAD

**Issue by nils-wisiol** [nils-wisiol/m4sc#18]
When running `python3 -m m4sc test GS --target qemu`, this happens:

```
Building qemu ... Build failed!
make VICTIM_NAME=GS VICTIM_TARGET=qemu
BUILD_DIR is not set. This should be an error, but is an info for now
BUILDING: qemu target
  LD    bin/qemu.elf
/usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: bin/bin/main.o: in function `recv_plain':
/home/nils/git/m4sc/victims/schoolbook32/main.c:48: undefined reference to `textbook'
collect2: error: ld returned 1 exit status
make: *** [target/stm32f4-discovery/rules.mk:142: bin/qemu.elf] Error 1
```

maybe there are textbook/schoolbook leftovers where there shouldn't be?

**Comment by nils-wisiol** [comment]
... tested on ee74ffd9d5e30fbda25b6df213b0aa6073654ecc

**Comment by hnj2** [comment]
Hey ho, I've included @TeumessianFox because i basically use the stm32... targets `rules.mk` file to build.
The problem is that both `victims/GS/main.c` and `victims/schoolbook32/main.c` are compiled to the location `bin/bin/main.o`.
In the case that `schoolbook32` was compiled after the last change to `victims/GS/main.c`, this leads to the GS binary being linked with the `victims/schoolbook32/main.c` file instead, which is the problem that we see here.

This will be fixed as soon as we implement the changes we discussed in our call.

As a workaround i suggest running `rm -rf ./bin` before `python -m m4sc test GS --target qemu`, which works on my machine.
